dot/network/service.go

```diff
@@ -33,18 +33,19 @@ var _ services.Service = &Service{}
 
 // Service describes a network service
 type Service struct {
-	ctx         context.Context
-	cfg         *Config
-	host        *host
-	mdns        *mdns
-	status      *status
-	gossip      *gossip
-	msgRec      <-chan Message
-	msgSend     chan<- Message
-	noBootstrap bool
-	noMdns      bool
-	noStatus    bool // internal option
-	noGossip    bool // internal option
+	ctx               context.Context
+	cfg               *Config
+	host              *host
+	mdns              *mdns
+	status            *status
+	gossip            *gossip
+	msgRec            <-chan Message
+	msgSend           chan<- Message
+	noBootstrap       bool
+	noMdns            bool
+	noStatus          bool            // internal option
+	noGossip          bool            // internal option
+	requestedBlockIDs map[uint64]bool // track requested block id messages
```

Comment:
fine for now, but let's consolidate this with the requestedBlockIDs in core soon.

Reply:
sure, any ideas ?

Reply:
yes, move this to state.Network and add functions for reading/writing from the mapping to the NetworkState interface

Reply:
#653

Reply:
great